src/runtime_src/core/common/memaccess.cpp

```diff
@@ -165,12 +167,24 @@ perform_memory_action(xrt_core::device* device, xrt_core::aligned_ptr_type& buf,
 
     // Update the buffer index based on how far we have read
     void* current_buffer_location = static_cast<char *>(buf.get()) + bytes_seen;
+    boost::format err_fmt("%s: Code : %d - %s %u bytes from %s(0x%x)");
+    err_fmt % __func__;
```

Comment:
Hmm...does this like of code doing anything?

Looking at the code I see you are expecting that the first argument is expecting the name of the function.   Have you tested it to see if it actuals prints the expected message.

If it does work, then I learned something new today and would like to understand more on how this works.

Reply:
It does work! I found it to be a neat trick when the message being generated used the same string. One issue I have is (as far as I know) the parameters need to be placed in order.
```
dbenusov@xsjsagarw50:/proj/rdi/staff/dbenusov$ xbutil advanced --read-mem --address 0x800000000 --size 1024 -o /tmp/test.txt --force -d 17:00
ERROR: perform_memory_action: Invalid argument - reading 1024 bytes from DDR[0](0x800000000)
```

Reply:
Time for me to be educated with regards to this syntax.